packages/tablev2/Util.tsx

```diff
@@ -0,0 +1,161 @@
+import * as React from "react";
+import { css } from "emotion";
+import { Icon } from "../icon";
+import { SystemIcons } from "../icons/dist/system-icons-enum";
+import * as dt from "../design-tokens/build/js/designTokens";
+import * as style from "./style";
+import { RESIZE_ICON_SIZE } from "./style";
+import { Flex, FlexItem } from "../styleUtils/layout";
+import { Tooltip } from "../tooltip";
+import { textTruncate } from "../shared/styles/styleUtils";
+
+///////////////////////////////////////////////////////////////////////////////
+//                                 DRAGGABLE                                 //
+///////////////////////////////////////////////////////////////////////////////
+
+let initialX: number = 0;
+export const className = css`
+  padding: 0 0.25em;
+  cursor: col-resize;
+  display: flex;
+  align-items: center;
+  height: 100%;
+  z-index: 10;
+  position: absolute;
+  top: 0;
+  right: 0;
+  background-color: ${dt.themeBgPrimary};
+  opacity: var(--draggable-opacity);
+  &:hover {
+    background-color: ${dt.themeBgHover};
+  }
+`;
+
+type E = React.MouseEvent;
+/**
+ * The draggable small handler used to size table columns.
+ */
+export const Draggable = (props: {
+  onRelativeXChange: (i: number) => void;
+}) => {
+  const [x, setX] = React.useState(0);
+
+  const updateMousePosition = (e: MouseEvent) => {
+    setX(e.clientX - initialX);
+  };
+
+  const onMouseUp = (e: MouseEvent) => {
+    props.onRelativeXChange(e.clientX - initialX);
+    setX(0);
+    document.removeEventListener("mousemove", updateMousePosition);
+    document.removeEventListener("mouseup", onMouseUp);
+  };
+  const onMouseDown = (e: E) => {
+    initialX = e.clientX;
+    document.addEventListener("mousemove", updateMousePosition);
+    document.addEventListener("mouseup", onMouseUp);
+  };
+
+  return (
+    <div
+      onMouseDown={onMouseDown}
+      className={className}
+      style={{ transform: `translate(${x}px, 0)` }}
+      role="button"
+    >
+      <div>
+        <Icon
+          shape={SystemIcons.ResizeHorizontal}
+          size={RESIZE_ICON_SIZE}
+          color={dt.textColorSecondary}
+        />
+      </div>
+    </div>
+  );
+};
+
+///////////////////////////////////////////////////////////////////////////////
+//                            HEADER CELL CONTENT                            //
+///////////////////////////////////////////////////////////////////////////////
+export const HeaderCellContent = ({
```

Comment:
@mperrotti could we follow this "composability over configuration" school of thought here to remove complexity from the table itself by handing users a `tooltipify`-function or a `TooltipCell`-component or something like that?

Reply:
You're right, "composability over configuration" is the way to go 👍